src/cargo/command_bridge.rs

```diff
@@ -0,0 +1,141 @@
+//! This struct is a simple bridge that we can inspect for test purpose
+
+use std::collections::HashMap;
+use std::convert::{AsRef, Into};
+use std::default::Default;
+use std::ffi::{OsStr, OsString};
+use std::process::{Command, Stdio};
+
+/// This temporary struct is used to inspect the generated command before calling the process::Command
+/// builder, as we can't inspect its structure and make assertions on it
+#[derive(Debug, Default)]
+pub struct CommandBridge {
+    pub name: String,
+    pub args: Vec<OsString>,
+    pub env: HashMap<OsString, OsString>,
+    pub stdout: Option<Stdio>,
+    pub stderr: Option<Stdio>,
+}
+
+impl CommandBridge {
+    pub fn new(name: &str) -> Self {
+        CommandBridge {
+            name: String::from(name),
+            .. Default::default()
+        }
+    }
+
+    pub fn arg<S: AsRef<OsStr>>(&mut self, arg: S) -> &mut Self {
+        let mut os_string = OsString::new();
+        os_string.push(arg);
+        self.args.push(os_string);
+        self
+    }
+
+    pub fn env<S: AsRef<OsStr>>(&mut self, key: S, value: S) -> &mut Self {
+        let mut key_str = OsString::new();
+        let mut value_str = OsString::new();
+
+        key_str.push(key.as_ref());
+        value_str.push(value.as_ref());
+
+        self.env.insert(key_str, value_str);
```

Comment:
You can just use [`to_os_string`]( https://doc.rust-lang.org/std/ffi/struct.OsStr.html#method.to_os_string)

Reply:
ooh, I missed that. I am not familiar with the `OsStr(ing)?` interface

Reply:
People tend to miss it. The Standard Library docs are great! Usually I'll take a look before using something to see what's available, especially if I tend to do common patterns like conversions!